crates/test/src/rt/mod.rs

```diff
@@ -20,6 +26,15 @@ pub struct Context {
     current_log: RefCell<String>,
     current_error: RefCell<String>,
     ignore_this_test: Cell<bool>,
+    formatter: Box<Formatter>,
+}
+
+trait Formatter {
```

Comment:
Why not just use `std::io::Write` and `writeln!(..)`?

Reply:
Hm perhaps! I think it's too much abstraction though because we can't generate any errors here and it'd require utf8 parsing and whatnot. This isn't really reused anywhere so we shouldn't lose too much I think without the trait implementation.